bundles/org.dataflowanalysis.converter/src/org/dataflowanalysis/converter/PCMConverter.java

```diff
@@ -342,21 +428,424 @@ private Label getOrCreateDFDLabel(CharacteristicValue charValue) {
 
         return label;
     }
+    
+    /**
+     * Get or create a Label with the supplied String name and LabelType
+     * @param labelName the name of the Label
+     * @param type the LabelType
+     * @return the Label
+     */
+    private Label getOrCreateDFDLabel(String labelName, LabelType type) {
+        Label label = type.getLabel()
+                .stream()
+                .filter(f -> f.getEntityName()
+                        .equals(labelName))
+                .findFirst()
+                .orElseGet(() -> createLabel(labelName, type));
 
+        return label;
+    }
+
+    /**
+     * Creates a Label from the supplied CharacteristicValue and adds it to the LabelType
+     * @param charValue the CharacteristicValue
+     * @param type the LabelType
+     * @return the created Label
+     */
     private Label createLabel(CharacteristicValue charValue, LabelType type) {
         Label label = datadictionaryFactory.eINSTANCE.createLabel();
         label.setEntityName(charValue.getValueName());
         type.getLabel()
                 .add(label);
         return label;
     }
+    
+    /**
+     * Creates a Label with the supplied name and adds it to the LabelType
+     * @param name the name of the Label
+     * @param type the LabelType
+     * @return the created Label
+     */
+    private Label createLabel(String name, LabelType type) {
```

Comment:
Why is this second createLabel function needed? It has the exact same functionality as the one above

Reply:
This creates a label, think `EU` or `Sensitive`, whilst the other creates a label of a label type, think `Location` or `Sensitivity`

Reply:
Fixed in 9161111817fd